documentation/modules/operators/con-operator-topic-names.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-topic-operator.adoc
+
+[id='con-operator-topic-names-{context}']
+= Topic naming conventions
+
+A `KafkaTopic` resource includes a name for the topic and a label that identifies the name of the Kafka cluster it belongs to. 
+
+.Label identifying a Kafka cluster for topic handling
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaTopicApiVersion}
+kind: KafkaTopic
+metadata:
+  name: topic-name-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  topicName: topic-name-1  
+----
+
+The cluster name is derived from the `Kafka` resource.
+If the label does not match the Kafka cluster, the Topic Operator cannot identify the `KafkaTopic`, and the topic is not created.
+
+Kafka and Kubernetes have their own naming validation rules, and a Kafka topic name might not be a valid resource name in Kubernetes.
+If possible, try and stick to a naming convention that works for both.
+
+Consider the following guidelines:
+
+* Use topic names that reflect the nature of the topic
+* Be concise and keep the name under 63 characters
+* Use all lower case and hyphens
+* Avoid special characters, spaces or symbols
+
+In the `KafkaTopic` resource, the Kafka topic name is specified using `metadata.name`.
+However, if the Kafka topic name is not a valid Kubernetes resource name, you can use the `spec.topicName` property as an alias that works in Kubernetes. 
+When absent, this name defaults to the `metadata.name` of the topic.
+Topic names cannot be changed later.
+
+.Example of supplying a valid Kafka topic name
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaTopicApiVersion}
+kind: KafkaTopic
+metadata:
+  name: My.Topic.1 # <1>
+spec:
+  topicName: my-topic-1 # <2>
+  # ...
+----
+<1> Kafka topic name uses upper case and periods, which are invalid in Kubernetes.
+<2> A topic name alias that works in Kubernetes.
```

Comment:
Isn't this the wrong way around? The `.metadata.name` has to be a valid Kubernetes name. `.spec.topicName` can contain the actualKafka name.

Reply:
Yes! I've updated and tried to make clearer.

> The `KafkaTopic` resource allows you to specify the Kafka topic name using the `metadata.name` field. 
> However, if the desired Kafka topic name is not a valid Kubernetes resource name, you can use the `spec.topicName` property to specify the actual name. 
> The `spec.topicName` field is optional, and when it's absent, the Kafka topic name defaults to the `metadata.name` of the topic.
> When a topic is created, the topic name cannot be changed later.
> 
> .Example of supplying a valid Kafka topic name
> [source,yaml,subs="attributes+"]
> apiVersion: {KafkaTopicApiVersion}
> kind: KafkaTopic
> metadata:
>   name: my-topic-1 # <1>
> spec:
>   topicName: My.Topic.1 # <2>
> <1> A valid topic name that works in Kubernetes.
> <2> A Kafka topic name that uses upper case and periods, which are invalid in Kubernetes.